Submitted By: Ryan Oliver <ryan.oliver@pha.com.au
Date: 2005-01-23
Initial Package Version: XFree 4.2.99
Upstream Status: Unknown
Origin: Gentoo 
Description: Ensures objects archived in static libs get built with -fPIC 
	A notable barf is when building libaudio (NAS) linking against
	libXau.a

--- xc/lib/Xau/Imakefile.IncludeSharedObjectInNormalLib	2001-01-17 14:42:24.000000000 -0500
+++ xc/lib/Xau/Imakefile	2003-01-29 02:25:49.000000000 -0500
@@ -37,6 +37,8 @@
 OBJS =	AuDispose.o AuFileName.o AuGetAddr.o AuGetBest.o AuLock.o \
 	AuRead.o AuUnlock.o AuWrite.o $(K5ENCOBJ)
 
+#define IncludeSharedObjectInNormalLib
+
 #include <Library.tmpl>
 
 DependTarget()
--- xc/lib/xkbfile/Imakefile.IncludeSharedObjectInNormalLib	2001-08-27 13:40:59.000000000 -0400
+++ xc/lib/xkbfile/Imakefile	2003-01-29 02:25:49.000000000 -0500
@@ -34,6 +34,8 @@
 
 ROOT_DEFINES = -DDFLT_XKB_CONFIG_ROOT=\"$(LIBDIR)/xkb\"
 
+#define IncludeSharedObjectInNormalLib
+
 #include <Library.tmpl>
 
 DependTarget()
--- xc/lib/xkbui/Imakefile.IncludeSharedObjectInNormalLib	1999-03-13 22:21:21.000000000 -0500
+++ xc/lib/xkbui/Imakefile	2003-01-29 02:25:49.000000000 -0500
@@ -15,6 +15,8 @@
          HEADERS = XKBui.h
         INCLUDES = -I. -I$(EXTINCSRC) -I../xkbfile
 
+#define IncludeSharedObjectInNormalLib
+
 #include <Library.tmpl>
 
 DependTarget()
--- xc/programs/Xserver/hw/xfree86/parser/Imakefile.IncludeSharedObjectInNormalLib	2003-01-29 02:41:13.000000000 -0500
+++ xc/programs/Xserver/hw/xfree86/parser/Imakefile	2003-01-29 02:41:53.000000000 -0500
@@ -36,9 +36,11 @@
 		 -DXCONFIGFILE=\"$(XCONFIGFILE)\" \
 		 -DXVERSION="$(XVERS)"
 
+#define IncludeSharedObjectInNormalLib
+
 #include <Library.tmpl>
 
-SpecialCObjectRule(scan,NullParameter,$(CONFIG_DEFINES) $(MODULEDEFINES) $(EXT_DEFINES))
+SpecialCObjectRule(scan,NullParameter,$(CONFIG_DEFINES) $(MODULEDEFINES) $(EXT_DEFINES) $(PICFLAGS))
 
 NormalProgramTarget(cpconfig,cpconfig.o $(OBJS),NullParameter,$(LOCAL_LIBRARIES),NullParameter)
 
